backend/typescript/services/implementations/volunteerService.ts

```diff
@@ -148,6 +166,29 @@ class VolunteerService implements IVolunteerService {
     return userVolunteerDTOs;
   }
 
+  async getCheckInsAndSchedules(
+    volunteerId: string,
+  ): Promise<(CheckInDTO | SchedulingDTO)[]> {
+    const checkIns = await (
+      await this.checkInService.getCheckInsByVolunteerId(volunteerId)
+    ).map((checkIn) => ({ ...checkIn, type: ShiftType.CHECKIN }));
+    const schedulings = await (
+      await this.schedulingService.getSchedulingsByVolunteerId(volunteerId)
+    ).map((scheduling) => ({ ...scheduling, type: ShiftType.SCHEDULING }));
+    const shifts = [...checkIns, ...schedulings].sort((a, b) => {
+      const date1 =
+        "startDate" in a
+          ? dayjs(a.startDate)
+          : getDateWithVolunteerTime(a.startTime, a.volunteerTime!);
+      const date2 =
+        "startDate" in b
+          ? dayjs(b.startDate)
+          : getDateWithVolunteerTime(b.startTime, b.volunteerTime!);
+      return date2.isBefore(date1) ? 1 : -1;
```

Comment:
`isBefore` returns a boolean but the javascript `sort` returns a number so it was a slight tweak to what was suggested